twoliter/src/cmd/publish_kit.rs

```diff
@@ -0,0 +1,66 @@
+use crate::project;
+use crate::tools::install_tools;
+use anyhow::{ensure, Context, Result};
+use clap::Parser;
+use std::path::PathBuf;
+use tokio::process::Command;
+
+/// Group all publish commands
+#[derive(Debug, Parser)]
+pub(crate) enum PublishCommand {
+    Kit(PublishKit),
+}
+
+impl PublishCommand {
+    pub(crate) async fn run(self) -> Result<()> {
+        match self {
+            PublishCommand::Kit(command) => command.run().await,
+        }
+    }
+}
+
+/// Publish a local kit to a container registry
+#[derive(Debug, Parser)]
+pub(crate) struct PublishKit {
+    /// Path to Twoliter.toml. Will search for Twoliter.toml when absent
+    #[clap(long = "project-path")]
+    project_path: Option<PathBuf>,
+    /// Kit name to build
+    kit_name: String,
+    /// Vendor to publish to
+    vendor: String,
+}
+
+impl PublishKit {
+    pub(super) async fn run(&self) -> Result<()> {
+        let project = project::load_or_find_project(self.project_path.clone()).await?;
+        let toolsdir = project.project_dir().join("build/tools");
+        install_tools(&toolsdir).await?;
+        let pubsys_path = toolsdir.join("pubsys");
+        let infra_path = project.project_dir().join("Infra.toml");
+        let kit_path = project
+            .project_dir()
+            .join("build")
+            .join("kits")
```

Comment:
I think this is simplified after a recent rebase:

```suggestion
        let kit_path = project
            .external_kits_dir()
```

Reply:
looks like it's slightly different, `external_kits_dir()` resolves to `build/external-kits`